packages/rum/src/domain/record/utils.ts

```diff
@@ -28,3 +28,19 @@ export function assembleIncrementalSnapshot<Data extends BrowserIncrementalData>
     timestamp: timeStampNow(),
   }
 }
+
+export function getPathToNestedCSSRule(rule: CSSRule): number[] {
+  const path: number[] = []
+  let currentRule = rule
+  while (currentRule.parentRule instanceof CSSGroupingRule) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** Do you really need to check the parentRule type? I would go with a simpler condition:

```suggestion
  while (currentRule.parentRule) {
```

We would some cast to make sure TS doesn't complain, but I think it's worth it. Especially when implementing support for browser who don't implement CSSGroupingRule

Reply:
I thought I removed this casting 😅